src/astro/databases/snowflake.py

```diff
@@ -57,6 +81,322 @@ def get_table_qualified_name(table: Table) -> str:  # skipcq: PYL-R0201
+        self.storage_integration = target_table.optional_args.get(
+            "storage_integration", None
+        )
+        if self.storage_integration is None:
+            raise Exception(
+                "To load dataset to snowflake using optimisation, specify the correct storage integration with correct"
+                " permission to the input file. Pass storage_integration={valid storage integration name} as "
+                "optional_args to output_table. "
+                "Read more: https://docs.snowflake.com/en/sql-reference/sql/create-storage-integration.html "
+                "Else disable optimisation by passing use_native_support=False in load_file."
+            )
+        method_name = self.native_support.get(source_file.location.location_type)
+        if method_name:
+            transfer_method = self.__getattribute__(method_name)
+            transfer_method(
+                source_file=source_file,
+                target_table=target_table,
+                if_exists=if_exists,
+                **kwargs,
+            )
+
+    def gs_to_snowflake(
+        self,
+        source_file: File,
+        target_table: Table,
+        if_exists: LoadExistStrategy = "replace",
+        **kwargs,
+    ) -> None:
+        """
+        Checks if the table and schema exists on Snowflake.
+        If table and schema exists, create stage based on integration passed by user.
+        Generate query and run COPY INTO command and return True else returns False
+
+        :param source_file: Source file
+        :param target_table: Output target table on snowflake
+        :param chunk_size: Chunk size for the file
+        :param if_exists: Update strategy for file
+        :param kwargs:
+        :return: None
+        """
+
+        # Create new table from file if snowflake table does not exists
+        if if_exists == "replace" and not self.table_exists(table=target_table):
+            self.drop_table(target_table)
+            self.create_emtpy_table(
+                source_file=source_file, target_table=target_table, if_exists=if_exists
+            )
+
+        # check if snowflake table exists
+        if self.table_exists(table=target_table):
+            logging.info("Table exists.")
+            stage_name = self.create_stage(source_file=source_file)
+            sql_parts = self.create_copy_into_sql_parts(
+                source_file=source_file, target_table=target_table, stage=stage_name
+            )
+            copy_query = "\n".join(sql_parts)
+
+            logging.info("Executing COPY command...")
+            execution_info = self.hook.run(
+                copy_query, autocommit=kwargs.get("autocommit", True)
+            )
+            logging.info("COPY command completed")
+            logging.info(execution_info)
+            self.drop_stage(stage_name=stage_name)
+
+    def create_stage(self, source_file: File) -> str:
```

Comment:
I separated this part in a PR of its own, with respective tests, and auxiliary methods for checking if a stage exists / to delete a stage:
https://github.com/astronomer/astro-sdk/pull/523